src/lseq/lseq.rs

```diff
@@ -360,7 +379,9 @@ impl<V: Ord + Clone + Display, A: Actor + Display> LSeq<V, A> {
                 // TODO: return error? the trait doesn't support that type of Result currently
                 panic!("p cannot be greater than q");
             } else if q_position > p_position {
-                q_position - p_position - 1
+                // (drusu) I'm having a hard time wraping my head around this
+                //         subtraction, is it valid to subtract these "paths"?
+                q_position - p_position
```

Comment:
They are not paths, they are the position the Identifiers (considering current depth/length) would occupy in the serialised sequence, i.e. position of p and q in the serialised sequence. This allows us to do this plain arithmetic, look at the LSeq paper in page 6 there is pseudo code of this. Once we know the diff of these positions is > 0 we know we can find a new number between them for the new Identifier, and at which depth, but we still don't know if they are occupied already by other Identifiers...which is what I still didn't cover here in the code.